x-pack/platform/plugins/shared/cloud/server/plugin.ts

```diff
@@ -264,9 +285,16 @@ export class CloudPlugin implements Plugin<CloudSetup, CloudStart> {
             })
           : undefined;
 
+        const queryResourceDataRaw = request.query?.resource_data ?? undefined;
+        const queryResourceData = queryResourceDataRaw
+          ? camelcaseKeys(queryResourceDataRaw, {
+              deep: true,
+            })
+          : undefined;
+
         const solutionType = this.config.onboarding?.default_solution;
 
-        if (queryOnboardingToken || queryOnboardingSecurity) {
+        if (queryOnboardingToken || queryOnboardingSecurity || queryResourceData) {
           core
             .getStartServices()
             .then(async ([coreStart]) => {
```

Comment:
nit: unrelated to this PR, but it appears as though `coreStart` is completely unused. Can we remove this whole `core.getStartServices().then...` part. And just have `persistTokenCloudData.catch(...)`

Reply:
@jloleysens, I though that since we are in the setup function we need to be in the start to access a savedObject type. I think we can remove the coreStart but we need to be in the start service to access the savedObject type. I think during setup we can register saved object type and that's it. Tell me if I am wrong here.

Reply:
Ah, thanks for pointing that out. Perhaps a code comment to that effect? Otherwise ignore this!

Reply:
Yes, I will update the PR to use coreStart for the saved object client that makes sense.